pic-sure-auth-services/src/main/java/edu/harvard/hms/dbmi/avillach/auth/service/MailService.java

```diff
@@ -20,53 +22,79 @@
 import java.io.FileReader;
 import java.io.IOException;
 import java.io.StringWriter;
-import java.io.Writer;
+import java.util.Map;
 
 public class MailService {
 	private static Logger logger = LoggerFactory.getLogger(MailService.class);
 	private static MustacheFactory mf = new DefaultMustacheFactory();
 
-	private static Mustache accessEmail = null;
-	
-	private void loadTemplates() {
-		try(FileReader reader = new FileReader(JAXRSConfiguration.userActivationTemplatePath);){
-			accessEmail = mf.compile(reader, "accessEmail");
+	private Mustache loadTemplates(String templateFile) throws IOException {
+		try(FileReader reader = new FileReader(JAXRSConfiguration.emailTemplatePath + templateFile)){
+			return mf.compile(reader, templateFile);
 		} catch (FileNotFoundException e) {
-			e.printStackTrace();
+			logger.error(e.getMessage(), e);
+			throw e;
 		} catch (IOException e) {
-			e.printStackTrace();
+			logger.error(e.getMessage(), e);
+			throw e;
 		}
 	}
 
+	/**
+	 * Send email to user about changes in user Roles
+	 * @param user
+	 */
 	public void sendUsersAccessEmail(User user){
-		if(accessEmail==null) {
-			loadTemplates();
+		if (user == null) {
+			logger.error("sendUsersAccessEmail(User) - User is null.");
```

Comment:
Here it would make sense to throw an exception. This method should never be called with a null user, so something upstream should have done that check or be prepared to handle the resulting exception.

Reply:
prior to calling this there is a (user instanceof User) check happening. So NPE will never be thrown in this case